src/coreclr/vm/stubgen.cpp

```diff
@@ -2413,7 +2462,26 @@ ILStubLinker::ILStubLinker(Module* pStubSigModule, const Signature &signature, S
             m_iTargetStackDelta--;
         }
 
-        m_nativeFnSigBuilder.SetCallingConv((CorCallingConvention)uNativeCallingConv);
+        if (m_nativeFnSigBuilder.GetCallingConv() == IMAGE_CEE_CS_CALLCONV_UNMANAGED)
+        {
+            switch((CorUnmanagedCallingConvention)uNativeCallingConv)
+            {
+                case IMAGE_CEE_CS_CALLCONV_C:
+                case IMAGE_CEE_CS_CALLCONV_STDCALL:
+                case IMAGE_CEE_CS_CALLCONV_THISCALL:
+                case IMAGE_CEE_CS_CALLCONV_FASTCALL:
+                    m_nativeFnSigBuilder.AddCallConvModOpt(GetToken(GetModOptTypeForCallConv((CorUnmanagedCallingConvention)uNativeCallingConv)));
+                    break;
+                default:
+                    // Ignore the Unmanaged callconv since it already has the modopts we need.
+                    // Ignore other callconvs since we'll set the right one later.
```

Comment:
>we'll set the right one later

Any way we can point to the phase or general part of the process where this will occur. These "later" statements are difficult to debug and even harder to parse as comments age.